documentation/modules/oauth/proc-oauth-client-config.adoc

```diff
@@ -28,46 +57,101 @@ Add a client callback plugin to your _pom.xml_ file, and configure the system pr
 +
-For example:
+For example, we can add the following to a `client.properties` file:
 +
-[source,env, subs="+quotes,attributes"]
+--
+.Client credentials mechanism properties 
+[source,properties,subs="+quotes,attributes"]
 ----
-System.setProperty(ClientConfig.OAUTH_TOKEN_ENDPOINT_URI, “_https://<auth-server-address>/auth/realms/master/protocol/openid-connect/token_”); <1>
-System.setProperty(ClientConfig.OAUTH_CLIENT_ID, "_<client_name>_"); <2>
-System.setProperty(ClientConfig.OAUTH_CLIENT_SECRET, "_<client_secret>_"); <3>
-System.setProperty(ClientConfig.OAUTH_SCOPE, "_<scope_value>_") <4>
-System.setProperty(ClientConfig.OAUTH_AUDIENCE, "_<audience_value_") <5>
+security.protocol=SASL_SSL # <1>
+sasl.mechanism=OAUTHBEARER # <2>
+sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
+  oauth.token.endpoint.uri="_<token_endpoint_url>_" \ # <3>
+  oauth.client.id="_<client_id>_" \ # <4>
+  oauth.client.secret="_<client_secret>_" \ # <5> 
+  oauth.scope="_<scope>_" \ # <6> 
+  oauth.audience="_<audience>_" ; # <7> 
 ----
-<1> URI of the authorization server token endpoint.
-<2> Client ID, which is the name used when creating the _client_ in the authorization server.
-<3> Client secret created when creating the _client_ in the authorization server.
-<4> (Optional) The `scope` for requesting the token from the token endpoint.
+<1> `SASL_SSL` security protocol for TLS-encrypted connections. Use `SASL_PLAINTEXT` over unencrypted connections for local development only.
+<2> The SASL mechanism specified as `OAUTHBEARER` or `PLAIN`. 
+<3> URI of the authorization server token endpoint.
+<4> Client ID, which is the name used when creating the _client_ in the authorization server.
+<5> Client secret created when creating the _client_ in the authorization server.
+<6> (Optional) The `scope` for requesting the token from the token endpoint.
 An authorization server may require a client to specify the scope.
-<5> (Optional) The `audience` for requesting the token from the token endpoint.
+<7> (Optional) The `audience` for requesting the token from the token endpoint.
 An authorization server may require a client to specify the audience.
-
-. Enable the _OAUTHBEARER_ or _PLAIN_ mechanism on a TLS encrypted connection in the Kafka client configuration.
+--
 +
-For example:
+--
+.Password grants mechanism properties 
+[source,properties,subs="+quotes,attributes"]
+----
+security.protocol=SASL_SSL
+sasl.mechanism=OAUTHBEARER
+sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
+  oauth.token.endpoint.uri="_<token_endpoint_url>_" \
+  oauth.client.id="_<client_id>_" \ # <1>
+  oauth.client.secret="_<client_secret>_" \ # <2>
+  oauth.password.grant.username="_<username>_" \ # <3> 
+  oauth.password.grant.password="_<password>_" \ # <4> 
+  oauth.scope="_<scope>_" \
+  oauth.audience="_<audience>_" ;
```

Comment:
Maybe add comments here like in line 78.

Reply:
Not sure we need to repeat. We haven't repeated the callouts of `security.protocol` or `sasl.mechanism`